api/service/v1alpha1/service.proto

```diff
@@ -85,6 +85,13 @@ service KargoService {
   rpc ListCredentials(ListCredentialsRequest) returns (ListCredentialsResponse);
   rpc UpdateCredentials(UpdateCredentialsRequest) returns (UpdateCredentialsResponse);
 
+  /* Secrets APIs */
+
+  rpc ListSecrets(ListSecretsRequest) returns (ListSecretsResponse);
+  rpc CreateSecrets(CreateSecretsRequest) returns (CreateSecretsResponse);
+  rpc UpdateSecrets(UpdateSecretsRequest) returns (UpdateSecretsResponse);
+  rpc DeleteSecrets(DeleteSecretsRequest) returns (DeleteSecretsResponse);
```

Comment:
I think these sound better if they are singular instead of plural. These are, after all creating, updating, or deleting a single k8s Secret resource.

(I am guessing you pluralized these initially to match CreateCredentials, UpdateCredentials, etc., credentials is a peculiar word because it's common in English to use "credentials" even when talking about only one. That doesn't apply to secret/secrets.)

```suggestion
  rpc CreateSecret(CreateSecretRequest) returns (CreateSecretResponse);
  rpc UpdateSecret(UpdateSecretRequest) returns (UpdateSecretResponse);
  rpc DeleteSecret(DeleteSecretRequest) returns (DeleteSecretResponse);
```